generators/csharp/codegen/src/ast/Field.ts

```diff
@@ -137,19 +155,31 @@ export class Field extends AstNode {
         if (this.static_) {
             writer.write("static ");
         }
+        if (this.readonly) {
+            writer.write("readonly ");
+        }
         writer.writeNode(this.type);
         writer.write(` ${this.name}`);
 
         const useExpressionBodiedPropertySyntax = this.get && !this.init && !this.set && this.initializer != null;
         if ((this.get || this.init || this.set) && !useExpressionBodiedPropertySyntax) {
             writer.write(" { ");
             if (this.get) {
+                if (this.get !== true && this.access !== this.get) {
```

Comment:
I think we can make this a little more clear - it seems like the main goal is to confirm `this.get` is not a boolean right? If not, that's when we write the access.

How about something like this instead?

```suggestion
                if (typeof this.get !== "boolean") {
                  writer.write(`${this.get} `);
                }
```

At that point, we could even include a helper function local to this file (until it's required elsewhere). That way we can just do:

```typescript
if (!isBoolean(this.get)) {
  writer.write(`${this.get} `);
}
```

Reply:
How about this?
```ts
    private hasSameAccess(access: Access | boolean): boolean {
        return access === true || access === this.access;
    }
    
            if (this.get) {
                if (!this.hasSameAccess(this.get)) {
                    writer.write(`${this.get} `);
                }
                writer.write("get; ");
            }
            if (this.init) {
                if (!this.hasSameAccess(this.init)) {
                    writer.write(`${this.init} `);
                }
                writer.write("init; ");
            }
            if (this.set) {
                if (!this.hasSameAccess(this.set)) {
                    writer.write(`${this.set} `);
                }
                writer.write("set; ");
            }
    
    ```